### PR TITLE
Milestone 5

### PR DESCRIPTION
**Pull request by MiguelArgentina**
- The user see all users list.
- A user can see other users page with their name and posts
- Created model with associations
- Created users views for index & show
- A user can send a friendship invitation.
- A user can see a button “Invite to friendship” next to other users name
- A user can see pending friendship invitations sent from other users
- A user can accept or reject friendships invitations

### CODE REVIEW

File: app/views/users/_user.html.erb [hunk 1]
```diff
@@ -1,7 +1,19 @@
...
+    &nbsp;
+    <%= link_to 'See Profile', user_path(user), class: 'profile-link' %>
+
+    <% unless a_rejected(user) %>
+      <% if a_requested(user) %>
+          &nbsp;|&nbsp; <%= link_to 'Accept friend request',  accept(user), class: 'profile-link' %>
+          &nbsp;|&nbsp; <%= link_to 'Reject friend request',  reject(user), class: 'profile-link' %>
+      <% elsif who_to_add(user) && not_pending(user) %>
+          &nbsp;|&nbsp; <%= link_to 'Invite to friendship',  users_path(user, friend_id: user.id), class: 'profile-link' %>
+      <% end %>
```
Comment (oloomoses) [inline app/views/users/_user.html.erb]: - [x] Kindly move this logic into helpers

File: app/views/users/show.html.erb [hunk 2]
```diff
@@ -3,10 +3,46 @@
     <%= gravatar_for @user %>
     <%= "Name: #{@user.name}" %>
   </h2>
+
+  <% if a_requested(@user) %>
+      <p>This user sent a friend request</p>
+      <%= link_to 'Actions',  user_path(current_user), class: 'action btn btn-warning' %>
+  <% elsif who_to_add(@user) && not_pending(@user) %>
+      <%= link_to 'Invite to friendship',  user_path(@user, friend_id: @user.id), class: 'action btn btn-secondary' %>
+  <% end %>
```
Comment (oloomoses) [inline app/views/users/show.html.erb]: - [x] Kindly move this logic into helpers

File: app/views/users/show.html.erb [hunk 3]
```diff
@@ -3,10 +3,46 @@
...
+        My Friends: <%= all_friends %>
+      </h5>
+      <% current_user.friends.each do |friend| %>
+        <div class="card-body">
+          <h6 class="card-title fw-bold">Name</h6>
+          <p class="card-text"><%= friend.name %></p>
+        </div>
+      <% end %>
+    </div>
+  <% end%>
```
Comment (oloomoses) [inline app/views/users/show.html.erb]: - [x] Kindly move this logic into helpers

File: app/views/layouts/application.html.erb [hunk 4]
```diff
@@ -17,7 +17,13 @@
         <div class="menu-item">
           <%= menu_link_to 'All users', users_path %>
         </div>
+        <% if current_user %>
+          <div class="menu-item">
+            <%= link_to 'My profile', current_user %>
+          </div>
+        <% end %>
```
Comment (oloomoses) [inline app/views/layouts/application.html.erb]: - [x] Kindly move this logic into helpers

**Review by oloomoses — CHANGES_REQUESTED** [review]
# Changes Required :recycle: 
Hello team, you have done an amazing job on this project. :muscle: 
You have worked on most of the features very well and only a few are yet to be implemented for this project to be approved. Kindly check my comments :point_down: 

### Highlights
- Well done on the linters and robocop configuration :100: 
- Well done on the pr, readme, and the commit messages :muscle:  They are descriptive, simple, and easy to follow :tada: 

### To Improve
- [x] Files that are not part of the project should be included in the .gitignore file. Kindly do this for the .idea file.
- [x] Posts timeline includes posts written by logged-in user and their friends. I should not see other people's posts if friendship is not accepted.
- [x] You have logic in your users and application views. Kindly move them into helpers. Views should strictly be used to render content from the corresponding controller and not to generate one.
- [x] Kindly add unit and integration tests for the app

**Kindly implement the above changes before submitting another review.**
**Feel free to reach me on slack or leave a comment while mentioning me and I will be happy to reply**

_Slack: Oloo Moses_

File: app/views/layouts/application.html.erb (hunk 4 above)
Comment (george-swift) [reply app/views/layouts/application.html.erb]: @oloomoses The `application.html.erb` file was inherited from the `Microverse ror-social-scaffold` repo. Lines 23 to 38 in the updated file contain logic specific to signing in a user, signing out a user, notices and alerts. We added lines 20 - 24 above to render content from the `#usershow` in the controller. The request to move this logic to helpers means previous work on this repo was not only incorrectly done but ignores the fact that the logic here calls a controller action.

**Review by smcommits — CHANGES_REQUESTED** [review]
# Status: Changes Required

### Hello, @MiguelArgentina and @george-swift! You have done a great job. Your application appears to be remarkable. To make this project even better, we still need to address an issue pointed out by the previous TSE. 

## Required Changes

#### You haven't addressed the following issue as the previous TSE requested. 

- [x] **Add unit testing**

Excellent job with writing the integration tests. It is also very important that you write unit tests for the User model. You can use [rspec rails documentation](https://relishapp.com/rspec/rspec-rails/docs) to understand more about how to write unit tests for rails. 

# 
#### Please reach out to me on Slack if you need any assistance or if I have overlooked something and made an unfair request.

**Review by NtwaliHeritier — APPROVED** [review]
### **STATUS: APPROVED✔✔✔**

Hello team,
Great job making the required changes.

You can merge✔✔👍👍

Happy coding💻💻

**Pull request merged**